### PR TITLE
Run on pull-requests too

### PR DESCRIPTION
**Pull request by inverse**
- Run build on PR too - before it was only on direct pushes and branches from master.
- Run tests

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
-  push
+  pull_request: ~
+  push:
+    branches: [master]
 
 jobs:
   build:
     runs-on: ubuntu-20.04
-    env:
-      GITHUB_ACCESS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (josegonzalez) [inline .github/workflows/build.yml:11]: I think we still need this?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -18,5 +18,3 @@ jobs:
         go-version: 1.15
     - name: Build
       run: make -j dist
-    - name: List output
-      run: ls -la build
```
Comment (josegonzalez) [inline .github/workflows/build.yml:22]: Why this change?

Comment (josegonzalez) [inline .github/workflows/build.yml:22]: Why this change?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -2,13 +2,13 @@
 name: build
 
 on:
-  push
+  pull_request: ~
+  push:
+    branches: [master]
```
Comment (josegonzalez) [inline .github/workflows/build.yml:7]: This drops the build for non-master branches...

File: .github/workflows/build.yml (hunk 2 above)
Comment (inverse) [reply .github/workflows/build.yml:22]: I can revert it, but didnt see what value it provide

File: .github/workflows/build.yml (hunk 3 above)
Comment (inverse) [reply .github/workflows/build.yml:7]: It should still work for pull requests - the reason for this restriction is otherwise you'll end up with two pipeline executions for pull requests branched from seatgeek master. I can put it back if you really want it.

**Comment by inverse** [comment]
@josegonzalez make the changes back 👍🏼

File: .github/workflows/build.yml (hunk 1 above)
Comment (inverse) [reply .github/workflows/build.yml:11]: I'm honestly not sure what you'll need this for here...

Comment (josegonzalez) [reply .github/workflows/build.yml:11]: Ah okay didn't realize something else built the image.
